### PR TITLE
Display the new policies on the site instead of linking to github

### LINKED ISSUE #306: Display the policies (general and technical) on the website
Don't just link to the github repo - render the markdown for the policies on the website.

### PR DESCRIPTION
**Pull request by levitte**
This adds the necessary files and procedures to generate HTML from policis

The checked out directories $(CHECKOUTS)/technical-policies/policies
and $(CHECKOUTS)/general-policies/policies contain files in markdown
format, which get converted to HTML using pandoc with the new template,
and appropriate dirdata.yaml.

The base infrastructure in policies/general and policies/technical is also
created, and the majority of the links in policies/sidebar.shtml are moved
to policies/general/sidebar.shtml and policies/general/dirdata.yaml.

Several old policy files are dropped in favor of those generated in those
generated in policies/general/ and policies/technical/.  Some older policies
still remain, pending replacements.

Fixes #306

### CODE REVIEW

**Comment by levitte** [comment]
I've had ~2018bda3665b5874316e6deac35e8dcae6367559~ffd1977a12e823661da018fa3c662155db9cdb0f (got some necessary fixes) sitting in my local repo for a couple of years, waiting to get out there.  The changed bin/md-to-html5 makes it much easier to generated a properly framed web page from more or less any Markdown file.﻿

**Comment by levitte** [comment]
https://beta.openssl.org/ has a run of this PR.  Everything isn't quite right, but close

**Comment by paulidale** [comment]
Aren't the new policies meant to completely replace the old?  The beta doesn't look like that (& to be honest, we're not there yet).

**Comment by levitte** [comment]
> Aren't the new policies meant to completely replace the old? The beta doesn't look like that (& to be honest, we're not there yet).

We can't *remove* policies without a vote, can we?  So, those that don't have a replacement in general-policies or technical-policies must remain until such a replacement is in place.  I assume that's going to be an incremental thing.

However, to clarify, I've reworked the set of commits, so there are now separate once specifically to drop some of the old files in favor of their replacement.  I suggest doing that same thing in the future, every time a new replacement is added (merged), until we've cleared them all.

**Comment by levitte** [comment]
As seen by the new coding style, we could choose a better syntax hightlighting style.  https://www.garrickadenbuie.com/blog/pandoc-syntax-highlighting-examples/ contains a few examples that pandoc supports, default is said to be pygmets, which isn't suitable for a dark background...

**Comment by mspncp** [comment]
> Display the new policies on the site instead of linking to github

I really like this idea! 👍

**Comment by mspncp** [comment]
> > Display the new policies on the site instead of linking to github
> 
> I really like this idea! 👍

Would it be a great effort to add an automatically generated „View this page on GitHub“ link at the bottom of the generated page? This would be helpful if people want to suggest changes via an issue or pull request. (I‘ve seen such links e.g. on GitHub Pages.)

File: Makefile
Comment (paulidale) [inline Makefile]: An 'x' too many.

File: Makefile
Comment (paulidale) [inline Makefile:248]: Any reason for the second sed instead of sed -e ... -e ...?
Just stylistic?

File: policies/general/index.md
Comment (paulidale) [inline policies/general/index.md]: Colon at the end?

File: policies/technical/index.md
Comment (paulidale) [inline policies/technical/index.md]: Isn't this just `<br>`?

File: policies/technical/index.md
Comment (paulidale) [inline policies/technical/index.md]: Again a colon at the end to introduce the list properly.

**Review by paulidale — COMMENTED** [review]
Looking decent.

File: policies/technical/index.md
Comment (levitte) [reply policies/technical/index.md]: I still have XHTML in my fingers :wink:

This is permitted in HTML 5, though...

If you want to enjoy a discussion on the topic, I recommend
https://stackoverflow.com/questions/1946426/html-5-is-it-br-br-or-br

File: Makefile
Comment (levitte) [reply Makefile]: Fixed

File: Makefile
Comment (levitte) [reply Makefile:248]: Let's say that it's just stylistic...

File: policies/general/index.md
Comment (levitte) [reply policies/general/index.md]: We don't usually do that.  But a period at the very least, yeah

Comment (levitte) [reply policies/general/index.md]: Done

File: policies/technical/index.md
Comment (levitte) [reply policies/technical/index.md]: Or a period...

File: policies/technical/index.md
Comment (levitte) [reply policies/technical/index.md]: BTW, not my fault...  it's already there in the file this came from:

https://github.com/openssl/web/blob/68f71e63270307dde5343cfc5b0f31b49cbd3167/policies/otc-policies.html#L15-L16

**Comment by levitte** [comment]
All review comments addressed so far

File: policies/technical/index.md
Comment (paulidale) [reply policies/technical/index.md]: You added a space, so it's obviously your fault :)

File: policies/technical/index.md
Comment (paulidale) [inline policies/technical/index.md:16]: Would this be better referencing the glossary?

[project bylaws]: ../glossary.html#bylaws

Comment (levitte) [reply policies/technical/index.md:16]: ....  so any link within policies/ will have to go through the glossary?  Not sure I agree with the forced indirection

Comment (levitte) [reply policies/technical/index.md:16]: I can, though, if you insist

Comment (paulidale) [reply policies/technical/index.md:16]: This was the only one that stood out as possibly benefiting from the indirection.  I'm not going to insist however.

The side bar will still have a direct link I believe.

**Comment by levitte** [comment]
So, final review?

**Review by paulidale — APPROVED** [review]
Another set of eyes would be good ...

**Comment by levitte** [comment]
Ping

**Comment by levitte** [comment]
Rebased to fix conflicts

**Comment by levitte** [comment]
Ping!

File: Makefile
Comment (mattcaswell) [inline Makefile]: Should be general-policies

File: Makefile
Comment (t8m) [inline Makefile]: This does not make sense to me as your're filtering out README.md from general policies from technical policies which does not contain the README.md at all but if it was there it would have technical-policies in the checkout path.

File: Makefile
Comment (mattcaswell) [inline Makefile:247]: Side note (not for this PR): This makes me wonder whether the original policies should actually be updated to reference the website directly rather than github.

File: bin/md-to-html5
Comment (mattcaswell) [inline bin/md-to-html5]: Typo: with with

File: bin/md-to-html5
Comment (mattcaswell) [inline bin/md-to-html5:39]: Perhaps add that -o is not allowed in this usage.

File: policies/technical/index.md
Comment (t-j-h) [inline policies/technical/index.md]: There is no point in showing First issued and Moved to GitHub here - this is now just a list - so no need for an attempt at some form of edit history for the *list*.

**Review by t-j-h — APPROVED** [review]

**Review by t8m — CHANGES_REQUESTED** [review]
Just one minor thing found.

**Comment by t-j-h** [comment]
Approved with or without fixing up the nit raised (dropping pointless date stuff).

I also agree with @t8m feedback

File: Makefile
Comment (t-j-h) [reply Makefile]: That should be fixed.

Comment (levitte) [reply Makefile]: Fixed by changing `general-policies` to `technical-policies`.  Other than that, I'd prefer to keep this, just in case someone adds a README.md at some point

File: policies/technical/index.md
Comment (levitte) [reply policies/technical/index.md]: I have zero issues dropping that.  Done

**Comment by levitte** [comment]
Review comments addressed

**Comment by levitte** [comment]
@t8m, satisfied?

File: bin/mk-md-titlelist
Comment (mattcaswell) [inline bin/mk-md-titlelist:31]: Can this ever happen given we have a default value for `$title`?

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: Makefile
Comment (levitte) [reply Makefile:247]: Nah.  If you look at them on github, you probably want to continue walking around on github

File: bin/mk-md-titlelist
Comment (levitte) [reply bin/mk-md-titlelist:31]: Yes.  If a markdown file starts with a line of equal signs, `$lastline` will be `undef`, and `$title` will therefore become `undef`

File: Makefile
Comment (levitte) [reply Makefile]: Fixed

File: bin/md-to-html5
Comment (levitte) [reply bin/md-to-html5:39]: Fixed

File: bin/md-to-html5
Comment (levitte) [reply bin/md-to-html5]: Fixed

**Review by t-j-h — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]

**Comment by levitte** [comment]
cfa00986aa96179b40b33e04d04b461132a5358a Rework the markdown to HTML script and template
24a70808948946553f92e62aa6553f2217b474c5 New script bin/mk-md-titlelist, for enumerating markdown files by title
a6adc71de7fe7654e20025c24ccd3dd274d92269 Add the necessary files and procedures to generate HTML from policies
b500ce5c38b6944b0a8ca5250b641c9778053adf Drop policies/otc-policies.html in favor of policies/technical/
b2c7f88e0a1d3aae9cd9e8c1e2e8be501b232caf Drop policies/codingstyle.{txt,html} in favor of policies/technical/coding-style.html
9fcf5661d5c427c42537f28e20c3bf369115ea84 Drop policies/committers.html in favor of general/committer-policy.html
2321b204ef53c5a4503a818aede0401d69acbe72 Make an exception for the glossary
de5c548e809b92e172dfc353ed9e095a04a6e39f Convert policies/{general,technical}/index.html -> index.md
cd366c982506776b15325e039b9891fe3b6f702b Rework all links that point to General / Technical Policies on github.com
d4c7508c91045368714e318f0d0437fe4015499d inc/screen.css: don't modify the 'white-space' attribute on 'article a'
a53269e8550a5a5abacd5d826a7166729f59bd0a Generate inc/pandoc-template.html5
a2c6bd0c1de6c35a7cf92b85d61182cf4242a6c4 Use the solarized theme for code highlighting with pandoc

**Pull request closed without merging**
